CSS Flex Layout

kiviniklas opened the pull request:
## Warum das nötig ist

Das grundlegende CSS Layout von kivi verwendet position: absolute, um den Frame Header oben, das Menü, die Actionbar und den Hauptbereich zu positionieren. Dies macht es extrem schwer bis unmöglich, Platz für Bedienelemente bedarfsgerecht zu layouten.

Zwei Beispiele:
- Mehr Felder in der Schnellsuchleiste einstellen, als in eine Bildschirmzeile passen
- Menüvariante Oben (CSS) und hinreichend kleiner Bildschirm: lange Menüs laufen unten aus dem Viewport heraus
Beides funktioniert mit dem hier entwickelten Flex Layout bereits besser.

## Status Quo

So sieht es mit etwas herunter gescrolltem Menu und geschrollter Belegmaske aus:
<img width="957" height="307" alt="grafik" src="https://github.com/user-attachments/assets/80df2145-1162-4873-8f4f-b73e77c561ea" />
Aktuell scrollt die H1-Überschrift noch mit dem Beleg, während die ActionBar fest steht. Ziel ist, dass auch die H1-Überschrift wieder fest stehen bleiben soll.

## Nächste Schritte

- Menüvariante Oben (mit Javascript) zum Laufen bringen
- H1 muss, um nicht mit dem Hauptteil mitzuscrollen, in den Header wandern.
  - Entscheiden, ob H1 aus allen Templates entfernt werden kann und es genügt `$::form->{title}` immer als Text der H1 Überschrift einzubauen
- Testen
- evtl. bereits altes CSS aufräumen und alte Macken wie z.B. den fehlenden Zeilenumbruch des Menüs (Menüvariante Oben (mit CSS)) beheben

kiviniklas committed:
css flex layout: erster Entwurf - Menüvariante Oben (CSS) funktioniert

Behebt: das Menü scollt, wenn es größer als die Viewporthöhe ist.

- action bar Höhe richtig setzen, auch wenn Button ausgeklappt ist
- action bar float nach rechts

kiviniklas committed:
css flex layout: erster Entwurf - Menüvariante Links (CSS)

kiviniklas committed:
css flex layout: @content-margin (grauer Streifen unten) weg

rebootl commented:
Also ultra rudimentär ein Layout unter: SL/Layout/Title.pm
Mit:
```
package SL::Layout::Title;

use strict;
use parent qw(SL::Layout::Base);

sub pre_content {
  "<h1>" . $::locale->text($::form->{title} || "") . "</h1>\n";
}

1;
```

Und in: SL/Layout/V3.pm
Eingefügt:
```
use SL::Layout::Title;

sub init_sub_layouts {
  $_[0]->sub_layouts_by_name->{actionbar} = SL::Layout::ActionBar->new;
  $_[0]->sub_layouts_by_name->{flash}     = SL::Layout::Flash->new;

  [
    SL::Layout::None->new,
    SL::Layout::Top->new,
    SL::Layout::CssMenu->new,
    SL::Layout::Title->new,
    $_[0]->sub_layouts_by_name->{actionbar},
    $_[0]->sub_layouts_by_name->{flash},
    SL::Layout::Content->new,
  ]
}
```

Scheint bereits zu funktionieren:
<img width="2464" height="798" alt="image" src="https://github.com/user-attachments/assets/381df16b-6e84-42ad-bb95-b2db5294d484" />

Verstehe noch nicht ganz was da der unterschied ist zwischen sub_layouts_by_name und einfach direkt einfügen.

rebootl committed:
css flex layout: Layout für Seitentitel erstellt

rebootl committed:
css flex layout: Seitentitel Layout in Layout V3 eingebaut

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen 1/n, Einfacher Fall nur 'title'

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen 2/n, Fall 'title | html'

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen 3/n, Fall HTML.escape(title)

NOTE: Einige der Templates hatten überflüssige body und html tags, die
habe ich auch gleich entfernt.

rebootl commented:
Okay, habe hier mal angefangen die titel Elemente aus den Templates zu entfernen. Weiss gar nicht ob wir das hier in dem PR machen wollen aber die Änderung macht schon Sinn denke ich. Ansonsten kann man es auch wieder reverten. Das sind aber noch nicht alle Templates, das gibt schon noch einiges mehr zu tun. Insbesondere die dann wo der Titel noch im Controller angepasst werden muss.

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen 4/n, Fall FORM.title

rebootl committed:
SL::Presenter::Tag span_tag hinzugefügt

Analog zu div_tag

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen 5/n, Order

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen 6/n, DeliveryOrder

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen 7/n, Reclamation

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen: Korrektur Funktionsname/Kommentar

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen: Korrektur Admin Bereich vorerst nicht anpassen

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen 8/n, am

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen 9/n, bank_transactions/search

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen 10/n, cp (payment/receipt)

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen 11/n, ct (Ansprechpersonen Suche)

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen 12/n, chart/report_configuration_overview

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen 13/n, common/search_history

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen 14/n, bp/search (Print)

rebootl committed:
css flex layout: Titel Layout Zeilenumbruch hinzufügen

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen 15/n, custom_data_export_designer

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen 16/n, customer_vendor/form

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen 17/n, datev/export

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen 18/n, dbupgrade/..

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen: FIX remove call to deleted template

kiviniklas committed:
css flex layout: Menüvariante Oben (mit JS) ohne position:fixed

kiviniklas committed:
css flex layout: Menüvariante Oben (mit JS): Zeile umbrechen

kiviniklas committed:
css flex layout: Seitentitel Layout in Layout Javascript eingebaut

kiviniklas committed:
css flex layout: Seitentitel Layout in Layout Classic eingebaut

kiviniklas committed:
css flex layout: Seitentitel Layout in Layout Classic eingebaut (Selektor)

kiviniklas committed:
css flex layout: Platzierung ActionBar wiederhergestellt

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen 19/n, gl/search

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen 20/n, ic Preis Updates

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen 21/n, letter/load_drafts

+ falsche closing Tags entfernt

rebootl committed:
css flex layout: Seitentitel aus Templates entfernen: locales update

sschoeling commented on css/design40/less/menu.less:
```diff
@@ -227,37 +227,34 @@ li.sub {
 // Layout Web Page (LEFT & RIGHT AREA)
 // --------------------
 
+body > div.FB1 {
```
ein etwas sprechenderer Name für facebook1 wäre gut.

sschoeling reviewed (commented):
Sieht erstmal okay aus, getestet und bis auf die paar noch zu fixenden `<h1>` auch keine Fehler gesehen.

Zum Stil: Ich finde die zusätzlichen wrapper <divs> nicht schön, und würde die vermeiden wenn es geht. Beim Rest vom less/css kann ich nicht viel beisteuern.

kiviniklas committed:
css flex layout: Klassen sprechend benannt

kiviniklas committed:
S:L:Title: Style (kein pretty printing)

kiviniklas committed:
css flex layout: Seitentitel aus Templates entfernen 22/n, S:C:Part

kiviniklas committed:
css flex layout: Seitentitel aus Templates entfernen 23/n, Reconciliation

kiviniklas committed:
css flex layout: Seitentitel aus Templates entfernen 24/n, S:C:Reconciliation

kiviniklas committed:
css flex layout: Seitentitel aus Templates entfernen 25/n, S:ReportGenerator

kiviniklas committed:
css flex layout: Seitentitel aus Templates entfernen 26/n, SuSa

kiviniklas committed:
css flex layout: Seitentitel aus Templates entfernen 27/n, SEPA-Export-Details ansehen

kiviniklas committed:
css flex layout: Seitentitel aus Templates entfernen 28/n, ShopOrder list

kiviniklas committed:
css flex layout: Seitentitel aus Templates entfernen 29/n, TODO list

kiviniklas committed:
css flex layout: Seitentitel aus Templates entfernen 30/n, Finanzamt-Einstellungen

kiviniklas committed:
css flex layout: Seitentitel aus Templates entfernen 31/n, Bericht Lagerbuchungen

kiviniklas committed:
css flex layout: Seitentitel aus Templates entfernen 32/n, Bericht Lagerbestand

kiviniklas committed:
css flex layout: Seitentitel aus Templates entfernen 33/n, calculate_qty_dialog

Dialoge haben keine <h1>-Titel

kiviniklas committed:
css flex layout: Seitentitel aus Templates entfernen: locales Lauf

kiviniklas committed:
templates: oe/edit_periodic_invoices_config wird nur noch für Dialog verwendet

kiviniklas committed:
css flex layout: Seitentitel aus Templates entfernen 34/n, Letter load draft

kiviniklas committed:
Design FIX: Briefentwurf laden: wrapper div hinzugefügt

kiviniklas committed:
css flex layout: Seitentitel aus Templates entfernen 35/n, dbupgrade/..
